sycl/test/basic_tests/is_device_copyable.cpp

```diff
@@ -0,0 +1,58 @@
+// RUN: not %clangxx -fsycl -fsycl-device-only -fsyntax-only -Xclang -verify \
+// RUN: %s -I %sycl_include 2>&1 | FileCheck -check-prefix=CHECK-ERR %s
```

Comment:
Not sure I understand why the test is designed this way, could you please elaborate?
Can we just use `expected-warning` and `expected-error` comments for clang verify instead of using `FileCheck`?
Also, instead of marking some types as device copyable in a separate compilation can we just add duplicate types and have one compilation?
can we introduce additional types which will be

Reply:
expected-error is what I tried first, but that did not work as the error line number points to header, but 'expected-error' needs to list the line number where it is placed in is_device_copyable.cpp file, e.g.: expected-error@+1

Reply:
> Also, instead of marking some types as device copyable in a separate compilation can we just add duplicate types and have one compilation?

Yes, I'll do that.

Reply:
> expected-error is what I tried first, but that did not work as the error line number points to header, but 'expected-error' needs to list the line number where it is placed in is_device_copyable.cpp file, e.g.: expected-error@+1

FWIW, there are a bunch of features to the diagnostic verifier that can help with this: https://clang.llvm.org/doxygen/classclang_1_1VerifyDiagnosticConsumer.html (see the Detailed Description part).

So you can likely use something like `// expected-error@foo.h:15 {{whatever}}` to check the diagnostic instead.

Reply:
Yes, that is doable, but the line in that header may shift when unrelated changes are done there.

Reply:
There are more features to help with that. :-) You can create a "bookmark" that is used to match diagnostics. e.g.,
```
// Header.h
int whatever; // #SuperAwesomeError

// Source.cpp
foo(); // expected-error@#SuperAwesomeError
```
This pairs the expected error with the bookmark defined in the header with the special comment.

Reply:
Right. Adding a bookmark in sycl/include/CL/sycl/types.hpp does not seem reasonable.

Reply:
Fixed the test by adding duplicated types and removing one of compilation lines: 302ab3e79524cc9dc935af18e3949960470729f5

Reply:
> Right, but adding a bookmark in sycl/include/CL/sycl/types.hpp does not seem reasonable.

Ah, I think I've been caught out yet again by SYCL abusing lit tests a little bit. In lit, all the include files are part of the test set and so the expectation is that you can add your diagnostic checks to them. SYCL elected to do full end-to-end testing instead of something more cross-system-stable. :-(

For your patch, the current approach seems fine to me.